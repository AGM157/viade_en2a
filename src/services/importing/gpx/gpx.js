import { parseGpx, parseGpxFromFile } from "viade-gpx-parse";


function gpxTest() {
    let gpxString = '<?xml version="1.0" encoding="UTF-8" standalone="no" ?>'
    gpxString += '<gpx xmlns="http://www.topografix.com/GPX/1/1" creator="byHand" version="1.1" '
    gpxString += 'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
    gpxString += 'xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd">'

    gpxString += '<wpt lat="39.921055008" lon="3.054223107">'
    gpxString += '<ele>12.863281</ele>'
    gpxString += '<time>2005-05-16T11:49:06Z</time>'
    gpxString += '<name>Cala Sant Vicen√ß - Mallorca</name>'
    gpxString += '<sym>City</sym>'
    gpxString += '</wpt>'
    gpxString += '</gpx>'


    parseSuperString()
    
    const gpxObjetct = parseGpx(string, function(error, data) {
        console.log(error)
        console.log(data)
        console.log(data.metadata)
        console.log(data.tracks)
    })
    
    /*
    parseGpxFromFile("testFile.gpx", function (error, data) {
        console.log(error)
        console.log(data)
        console.log(data.metadata)
        console.log(data.tracks)
    })
    */

}






export default gpxTest;
