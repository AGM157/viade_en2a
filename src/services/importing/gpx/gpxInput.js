function parseSuperString() {
    var string = ""
    string += '<gpx creator="GPS Visualizer https://www.gpsvisualizer.com/" version="1.0">'
    string += '<wpt lat="45.44283" lon="-121.72904">'
    string += '<ele>1374'
    string += '</ele>'
    string += '<name>Vista Ridge Trailhead '
    string += '</name>'
    string += '<sym> Trail Head'
    string += '</sym>'

    string += '</wpt>'
    string += '<wpt lat="45.41000" lon="-121.71349">'
    string += '<ele>1777'
    string += '</ele>'
    string += '<name>Wy East Basin'
    string += '</name>'

    string += '</wpt>'
    string += '<wpt lat="45.41124" lon="-121.70404">'
    string += '<ele>1823'
    string += '</ele>'
    string += '<name>Dollar Lake'
    string += '</name>'

    string += '</wpt>'
    string += '<wpt lat="45.39260" lon="-121.69937">'
    string += '<ele>2394'
    string += '</ele>'
    string += '<name>Barrett Spur'
    string += '</name>'
    string += '<sym>Summit'
    string += '</sym>'

    string += '</wpt>'
    
    string += '<trk>'
    string += '<name>Barrett Spur 1'
    string += '</name>'
    string += '<desc>Uy como esto funcione xD'
    string += '</desc>'
    string += '<extensions>'
    string += '<line xmlns="http://www.topografix.com/GPX/gpx_style/0/2">'
    string += '<color>9900ff'
    string += '</color>'

    string += '</line>'

    string += '</extensions>'
    string += '<trkseg>'
    string += '<trkpt lat="45.4431641" lon="-121.7295456">'
    string += '<ele>1777'
    string += '</ele>'

    string += '</trkpt>'
    string += '<trkpt lat="45.4428615" lon="-121.7290800">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4425697" lon="-121.7279085">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4424274" lon="-121.7267360">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4422017" lon="-121.7260429">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4416576" lon="-121.7252347">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4406144" lon="-121.7241181">'
    string += '</trkpt>'
    string += '</trkseg>'



    string += '</trk>'
    string += '<trk>'
    string += '<name>Barrett Spur 2'
    string += '</name>'
    string += '<trkseg>'
    string += '<trkpt lat="45.3928201" lon="-121.6995658">'
    string += '</trkpt>'
    string += '<trkpt lat="45.3935449" lon="-121.6998805">'
    string += '</trkpt>'
    string += '<trkpt lat="45.3937897" lon="-121.6997710">'
    string += '</trkpt>'


    string += '</trkseg>'
    string += '<trkseg>'
    string += '<trkpt lat="45.4055556" lon="-121.7058619">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4057016" lon="-121.7055424">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4064672" lon="-121.7058247">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4065550" lon="-121.7056490">'
    string += '</trkpt>'
    


    string += '</trkseg>'
    string += '<trkseg>'
    string += '<trkpt lat="45.4099747" lon="-121.7134529">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4110685" lon="-121.7158641">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4121873" lon="-121.7171940">'
    string += '</trkpt>'
    string += '<trkpt lat="45.4131063" lon="-121.7179663">'
    string += '</trkpt>'



    string += '</trkseg>'

    string += '</trk>'

    string += '</gpx>'

    return string;

}

export default parseSuperString;