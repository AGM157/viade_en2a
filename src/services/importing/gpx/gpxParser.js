import { parseGpx, parseGpxFromFile } from "viade-gpx-parse";
import parseSuperString from "./gpxInput";
import { tracks } from "rdf-namespaces/dist/schema";
import {TrackPoint, Waypoint, Route} from "../../../domain/domainClasses";

export function gpxTest() {
    let gpxString = '<?xml version="1.0" encoding="UTF-8" standalone="no" ?>';
    gpxString += '<gpx xmlns="http://www.topografix.com/GPX/1/1" creator="byHand" version="1.1" ';
    gpxString += 'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ';
    gpxString += 'xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd">';

    gpxString += '<wpt lat="39.921055008" lon="3.054223107">';
    gpxString += '<ele>12.863281</ele>';
    gpxString += '<time>2005-05-16T11:49:06Z</time>';
    gpxString += '<name>Cala Sant Vicen√ß - Mallorca</name>';
    gpxString += '<sym>City</sym>';
    gpxString += '</wpt>';
    gpxString += '</gpx>';

    let string = parseSuperString()
    parseGpxToRoutes(string, function (routes) {
        console.log(routes);
    })
}

export function parseGpxToRoutes(gpxString, callback){

    parseGpx(gpxString, function (error, gpxData){
        var routes = [];
        var routeWaypoints = [];

        var waypoints = gpxData.waypoints;
        var tracks = gpxData.tracks;

        waypoints.forEach( (gpxWpt) => {
            routeWaypoints.push(new Waypoint(gpxWpt.name, gpxWpt.desc, gpxWpt.lat, gpxWpt.lon, gpxWpt.elevation));
        });

        tracks.forEach(track => {
            let route = parseRouteFromTrack(track);
            route.addWaypoints(routeWaypoints);       
            routes.push(route);
            
        });

        return callback(routes);
        
    })
}


function parseRouteFromTrack(track){
    var points = parsePointsOfTrack(track);

    let params = {
        name: track.name,
        description: track.description,
        waypoints: track.waypoints,
        itinerary: points
    };
    let testRoute = new Route(params);
    return testRoute;
}

function parsePointsOfTrack(track){
    var points = [];
    track.segments.forEach( (segment) => {
        segment.forEach(trackPoint => {
            points.push(new TrackPoint(trackPoint.lat, trackPoint.lon, trackPoint.elevation));
        });
    });
    return points;
}

function parseWaypointsOfGpx(Wpts){
    var waypoints = [];
    Wpts.forEach( (gpxWpt) => {
        waypoints.push(new Waypoint(gpxWpt.name, gpxWpt.desc, gpxWpt.lat, gpxWpt.lon, gpxWpt.elevation));
    });
    return waypoints;
}
export default gpxTest;