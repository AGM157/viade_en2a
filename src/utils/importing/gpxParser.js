import Comment from '../../domain/domainClasses';
import GPX from 'gpx-parser-builder';
import { RouteSpec, TrackPoint } from 'react-router-dom';


class GpxParser {

	/**
	 * Receives s STRING representing the file, and returns a domain route object
	 * @param {*} fileString the string representing the file.
	 * @returns A RouteSpec instance, from domainClasses.js
	 */
	parse(fileString) {
		const gpx = GPX.parse(fileString); //We get the gpx object with the library
		const listaRutas = [];
		const route = {};


		gpx.trk.forEach(trk => {
			route.name = trk.name;
			route.waypoints = [];
			trk.trkseg.forEach(trkseg => {
				trkseg.trkpt.forEach(trkpt => {
					route.waypoints.push(new TrackPoint(trkpt.name, trkpt.lat, trkpt.lng)
                        /*{ 
						"name":trkpt.name
						"lat":trkpt.lat,
						"lng":trkpt.lng,
                    }*/
					);
				});
			});
			listaRutas.push(new RouteSpec({
				name: route.name,
				itineray: route.waypoints
			}));
		});

		return listaRutas[0] //We are returning the first route, because of view problems. TECHNICAL DEBT
	}
}

export default gpxParser;


