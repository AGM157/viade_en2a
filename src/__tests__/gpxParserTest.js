import { parseGpxToRoutes } from "../services/importing/gpx/gpxParser";
import generateWaypointsInput from "../services/importing/gpx/testingFiles/gpxInput";
import generateNoTrackInput from "../services/importing/gpx/testingFiles/gpxNoTracksInput";
import parseSuperString from "../services/importing/gpx/gpxInput";
import "@testing-library/jest-dom/";

var routesArray = [];

function emptyCallback(routes) {
  console.log(routes);
  routesArray = routes;
}

test("Parsing two tracks", () => {
  parseGpxToRoutes(parseSuperString(), emptyCallback);
  expect(routesArray.length).toBe(2);
});

test("Parsing just one track", () => {
  parseGpxToRoutes(generateWaypointsInput(), emptyCallback);
  expect(routesArray.length).toBe(1);
});

test("Parsing no track", () => {
  console.log(generateNoTrackInput());
  parseGpxToRoutes(generateNoTrackInput(), emptyCallback);
  expect(routesArray.length).toBe(0);
});

test("Parsing two tracks snapshot", () => {
  parseGpxToRoutes(parseSuperString(), emptyCallback);
  expect(routesArray).toMatchSnapshot();
});

test("Parsing just one track snapshot", () => {
  parseGpxToRoutes(generateWaypointsInput(), emptyCallback);
  expect(routesArray).toMatchSnapshot();
});

test("Parsing no track snapshot", () => {
  parseGpxToRoutes(generateNoTrackInput(), emptyCallback);
  expect(routesArray).toMatchSnapshot();
});
