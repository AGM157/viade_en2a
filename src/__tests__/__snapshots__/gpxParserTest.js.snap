// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parsing just one track snapshot 1`] = `
Array [
  Route {
    "description": "Uy como esto funcione xD",
    "itinerary": Array [
      TrackPoint {
        "elevation": 1777,
        "latitude": 45.4431641,
        "longitude": -121.7295456,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4428615,
        "longitude": -121.72908,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4425697,
        "longitude": -121.7279085,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4424274,
        "longitude": -121.726736,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4422017,
        "longitude": -121.7260429,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4416576,
        "longitude": -121.7252347,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4406144,
        "longitude": -121.7241181,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4398193,
        "longitude": -121.722489,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4387649,
        "longitude": -121.7226112,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4383933,
        "longitude": -121.7224328,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.437785,
        "longitude": -121.7224159,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4372204,
        "longitude": -121.7226603,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4347837,
        "longitude": -121.7226007,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4332,
        "longitude": -121.721648,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4334576,
        "longitude": -121.7223143,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.432173,
        "longitude": -121.7222102,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4316609,
        "longitude": -121.7219974,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4303068,
        "longitude": -121.7220616,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4270753,
        "longitude": -121.7209685,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.426761,
        "longitude": -121.7211872,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4260133,
        "longitude": -121.7212623,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4257683,
        "longitude": -121.7214738,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.42574,
        "longitude": -121.7217762,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4259485,
        "longitude": -121.7226009,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4249972,
        "longitude": -121.7223672,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4246035,
        "longitude": -121.7219816,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4238682,
        "longitude": -121.721983,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4226721,
        "longitude": -121.7216494,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.422412,
        "longitude": -121.7217998,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4211497,
        "longitude": -121.7218767,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4193319,
        "longitude": -121.720865,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4186435,
        "longitude": -121.7202956,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4185934,
        "longitude": -121.7200745,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4178963,
        "longitude": -121.7196035,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4171101,
        "longitude": -121.7198115,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4166827,
        "longitude": -121.719325,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4161855,
        "longitude": -121.7190778,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4159291,
        "longitude": -121.7193146,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4153644,
        "longitude": -121.7193939,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4151268,
        "longitude": -121.7191578,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4148071,
        "longitude": -121.7191043,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.414631,
        "longitude": -121.7187962,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4142524,
        "longitude": -121.7187236,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4142844,
        "longitude": -121.7185595,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.413352,
        "longitude": -121.7180429,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4131406,
        "longitude": -121.7181383,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4130356,
        "longitude": -121.7179036,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4118436,
        "longitude": -121.7168789,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4109205,
        "longitude": -121.7156569,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4104523,
        "longitude": -121.714525,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.410493,
        "longitude": -121.7143814,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4102075,
        "longitude": -121.7140608,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4099806,
        "longitude": -121.7134527,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4099792,
        "longitude": -121.713461,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4091489,
        "longitude": -121.7134937,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4086133,
        "longitude": -121.7132504,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4080616,
        "longitude": -121.712767,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4076426,
        "longitude": -121.7126047,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4075043,
        "longitude": -121.7122301,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4070652,
        "longitude": -121.711898,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4068712,
        "longitude": -121.7114766,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4067987,
        "longitude": -121.7108634,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4064528,
        "longitude": -121.7106934,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4057286,
        "longitude": -121.7110326,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4056813,
        "longitude": -121.710828,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4055566,
        "longitude": -121.7109216,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4047244,
        "longitude": -121.7093884,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4039059,
        "longitude": -121.7083824,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4037176,
        "longitude": -121.7077738,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4034533,
        "longitude": -121.7074489,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4026499,
        "longitude": -121.7071945,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4019737,
        "longitude": -121.7067004,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4018086,
        "longitude": -121.7067477,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4014084,
        "longitude": -121.7063918,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4013177,
        "longitude": -121.7059701,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4011965,
        "longitude": -121.7058914,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4010688,
        "longitude": -121.7053257,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4008116,
        "longitude": -121.7054978,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4006075,
        "longitude": -121.7053495,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4005546,
        "longitude": -121.7054856,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3991622,
        "longitude": -121.7049765,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.398556,
        "longitude": -121.7042976,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3981831,
        "longitude": -121.704226,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3973151,
        "longitude": -121.7036992,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3967974,
        "longitude": -121.703637,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3963985,
        "longitude": -121.7033742,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3945456,
        "longitude": -121.7029688,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3920595,
        "longitude": -121.7015918,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3907614,
        "longitude": -121.7012029,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3906454,
        "longitude": -121.7010483,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3906726,
        "longitude": -121.7008185,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3909774,
        "longitude": -121.7008263,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3911315,
        "longitude": -121.70043,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3909963,
        "longitude": -121.6998193,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3908688,
        "longitude": -121.6997923,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3917895,
        "longitude": -121.6994679,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3926205,
        "longitude": -121.6994847,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3925915,
        "longitude": -121.6992485,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3928117,
        "longitude": -121.6995661,
      },
    ],
    "name": "Barrett Spur 1",
    "waypoints": Array [
      Waypoint {
        "description": undefined,
        "elevation": 1374,
        "latitude": 45.44283,
        "longitude": -121.72904,
        "name": "Vista Ridge Trailhead ",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1777,
        "latitude": 45.41,
        "longitude": -121.71349,
        "name": "Wy East Basin",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1823,
        "latitude": 45.41124,
        "longitude": -121.70404,
        "name": "Dollar Lake",
      },
      Waypoint {
        "description": undefined,
        "elevation": 2394,
        "latitude": 45.3926,
        "longitude": -121.69937,
        "name": "Barrett Spur",
      },
    ],
  },
]
`;

exports[`Parsing no track snapshot 1`] = `Array []`;

exports[`Parsing two tracks snapshot 1`] = `
Array [
  Route {
    "description": "Uy como esto funcione xD",
    "itinerary": Array [
      TrackPoint {
        "elevation": 1777,
        "latitude": 45.4431641,
        "longitude": -121.7295456,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4428615,
        "longitude": -121.72908,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4425697,
        "longitude": -121.7279085,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4424274,
        "longitude": -121.726736,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4422017,
        "longitude": -121.7260429,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4416576,
        "longitude": -121.7252347,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4406144,
        "longitude": -121.7241181,
      },
    ],
    "name": "Barrett Spur 1",
    "waypoints": Array [
      Waypoint {
        "description": undefined,
        "elevation": 1374,
        "latitude": 45.44283,
        "longitude": -121.72904,
        "name": "Vista Ridge Trailhead ",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1777,
        "latitude": 45.41,
        "longitude": -121.71349,
        "name": "Wy East Basin",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1823,
        "latitude": 45.41124,
        "longitude": -121.70404,
        "name": "Dollar Lake",
      },
      Waypoint {
        "description": undefined,
        "elevation": 2394,
        "latitude": 45.3926,
        "longitude": -121.69937,
        "name": "Barrett Spur",
      },
    ],
  },
  Route {
    "itinerary": Array [
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3928201,
        "longitude": -121.6995658,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3935449,
        "longitude": -121.6998805,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.3937897,
        "longitude": -121.699771,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4055556,
        "longitude": -121.7058619,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4057016,
        "longitude": -121.7055424,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4064672,
        "longitude": -121.7058247,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.406555,
        "longitude": -121.705649,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4099747,
        "longitude": -121.7134529,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4110685,
        "longitude": -121.7158641,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4121873,
        "longitude": -121.717194,
      },
      TrackPoint {
        "elevation": -1,
        "latitude": 45.4131063,
        "longitude": -121.7179663,
      },
    ],
    "name": "Barrett Spur 2",
    "waypoints": Array [
      Waypoint {
        "description": undefined,
        "elevation": 1374,
        "latitude": 45.44283,
        "longitude": -121.72904,
        "name": "Vista Ridge Trailhead ",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1777,
        "latitude": 45.41,
        "longitude": -121.71349,
        "name": "Wy East Basin",
      },
      Waypoint {
        "description": undefined,
        "elevation": 1823,
        "latitude": 45.41124,
        "longitude": -121.70404,
        "name": "Dollar Lake",
      },
      Waypoint {
        "description": undefined,
        "elevation": 2394,
        "latitude": 45.3926,
        "longitude": -121.69937,
        "name": "Barrett Spur",
      },
    ],
  },
]
`;
